src/GovUk.Education.ExploreEducationStatistics.Content.Services/DataSetFileService.cs

```diff
@@ -125,50 +121,54 @@ public class DataSetService : IDataSetService
             .ToListAsync();
     }
 
-    public async Task<Either<ActionResult, DataSetDetailsViewModel>> GetDataSet(
-        Guid releaseVersionId,
-        Guid fileId)
+    public async Task<Either<ActionResult, DataSetFileViewModel>> GetDataSetFile(
+        Guid dataSetId)
     {
-        // WARN: This prevents public uses from seeing data sets they shouldn't!
-        if (!await _releaseVersionRepository.IsLatestPublishedReleaseVersion(releaseVersionId))
+        var releaseFile = await _contentDbContext.ReleaseFiles
+            .Include(rf => rf.ReleaseVersion.Publication.Topic.Theme)
+            .Include(rf => rf.File)
+            .Where(rf =>
+                rf.File.DataSetFileId == dataSetId
+                && rf.ReleaseVersion.Published.HasValue
+                && DateTime.UtcNow >= rf.ReleaseVersion.Published.Value)
+            .OrderByDescending(rf => rf.ReleaseVersion.Version)
+            .FirstOrDefaultAsync();
```

Comment:
I've readded the `!await _releaseVersionRepository.IsLatestPublishedReleaseVersion` check so that we ensure we've fetched the latest release version for that particular release.

Just to make it crystal clear to all, this is how this could potentially have happend:
- Publish releases for 2000 and 2001 under the same publication, both with data sets added
- Create an amendment for the 2000 release
- Remove the data set
- Publish the 2000 release

As there is no ReleaseFile associated with the amendment, the code would fetch the release file for the original 2000 release version, not the amendment.

Reply:
Also added `DataSetFileRemovedOnAmendment_ReturnsNotFound` test.